statsrelay/src/backend_client/mod.rs

```diff
@@ -0,0 +1,18 @@
+use tokio::{sync::mpsc, task::JoinHandle};
+
+use crate::statsd_proto::Pdu;
+
+pub mod prom_client;
+pub mod statsd_client;
+
+pub trait BackendClient {
+    fn sender(&self) -> mpsc::Sender<Pdu>;
+
+    fn endpoint(&self) -> &str;
+
+    // Request the join handle for this client, which completes when all
+    // operations attempting to use this client are complete and client has shut
+    // down. Note that this is a destructive action, and a second call to a join
+    // handle will return None.
+    fn join_handle(&self) -> Option<JoinHandle<()>>;
```

Comment:
Def don't feel obligated to do this one, I'm just picking at JoinHandle appearing in a return position on a trait method since IME that particular idiom can lead to pretty verbose shutdown code for callers, if they want to gracefully drain things with a timeout.

Is there more to do with this joinhandle than just waiting for it to complete? If not, you might consider making the interface something like `async fn shutdown(self);` (or return an enum indicating state) and doing the flush/wait internal to this function. I know it'd mean pulling in async_trait and paying the extra indirection cost in the implementation, but it might be a little bit nicer for callers to interact with since it would encode the destructiveness in the type (it takes self by ownership).

Minor related nit: did you want this to be a doc-comment so it shows up in the rustdoc for this trait?